### PR TITLE
HotBackup documentation for 3.5

### PR DESCRIPTION
**Pull request by neunhoef**
Related PRs:
- https://github.com/arangodb/arangodb/pull/10056
- https://github.com/arangodb/arangodb/pull/9934
- https://github.com/arangodb/arangodb/pull/10055
- https://github.com/arangodb/arangodb/pull/9932

### CODE REVIEW

**Comment by Simran-B** [comment]
@sleto-it Turned out arangobackup is a C++ tool using the startup option framework, and the "verb"/"action" is actually an option `--operation` which is allowed to be passed as positional argument (without `--operation`).

I integrated the auto-generated table, which you can preview here:
https://deploy-preview-125--zealous-morse-14392b.netlify.com/docs/3.5/programs-arangobackup-options.html

Also opened the following related PRs (to be backported):
- https://github.com/arangodb/arangodb/pull/9934
- https://github.com/arangodb/arangodb/pull/9932

**Comment by sleto-it** [comment]
@Simran-B yay! thanks for the preview!

File: 3.5/backup-restore.md [hunk 1]
```diff
@@ -1,11 +1,22 @@
...
 
-Backup and restore can be done via the tools
+{%- comment %}
+TODO:
+- Introduction to topic
+- Offline dumps
+- Cluster
+{% endcomment %}
+
+Regular Backups
```
Comment (sleto-it) [inline 3.5/backup-restore.md]: - As discussed via phone, I would like to suggest the following changes
- We should have an introduction first, before the sections “regal backups” and “hot backups”
- In particular, the section “Performing frequent backups is important…” is general and not linked to regular backups
- Also elsewhere we call the backup done with arangodump “logical” - so perhaps we can use this name instead
- Also, version in which the hotbackup is introduced is missing
- Also, we could mention “cold” / raw backups (when server is down)
- Typo “restore” —> “restored”
- Missing introduction to what a logical backup is
- I have opened PR https://github.com/arangodb/docs/pull/167

Comment (sleto-it) [reply 3.5/backup-restore.md]: all handled but "Missing introduction to what a logical backup is" (lower prio)

**Comment by Simran-B** [comment]
I think we need to add two arangod option pages related to hot backup (from a different PR preview build):

```
docs/3.5/programs-arangod-options.html
   target does not exist --- docs/3.5/programs-arangod-options.html --> programs-arangod-backup.html
   target does not exist --- docs/3.5/programs-arangod-options.html --> programs-arangod-rclone.html
```

File: 3.5/programs-arangobackup-limitations.md [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+this limitation in one of the next releases.
+
+Note furthermore that a running query with views can prevent a restore
+operation from happening whilst the query is running.
+
+Windows not Supported Yet
+-------------------------
+
+The first release of the hot backup feature (3.5.1) does not support
+the Windows version of ArangoDB.
```
Comment (Simran-B) [inline 3.5/programs-arangobackup-limitations.md]: Is it planned for 3.5.2? If no, then we should avoid such a statement and just say that it's not supported yet.

File: 3.5/programs-arangobackup-limitations.md [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+Windows not Supported Yet
+-------------------------
+
+The first release of the hot backup feature (3.5.1) does not support
+the Windows version of ArangoDB. 
+
+Services on Single Server
+-------------------------
+
+On a single server the installed services are not backed up and are
```
Comment (Simran-B) [inline 3.5/programs-arangobackup-limitations.md]:
```suggestion
On a single server the installed Foxx microservices are not backed up and are
```

File: 3.5/programs-arangobackup-limitations.md [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+therefore also not restored. This is because in single server mode
+the service installation is done locally in the file system and does not
+track the information in the `_apps` collection.
+
+In a cluster, the coordinators will after a restore eventually restore
```
Comment (Simran-B) [inline 3.5/programs-arangobackup-limitations.md]:
```suggestion
In a cluster, the coordinators will eventually restore
```

File: 3.5/programs-arangobackup-limitations.md [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+the state of the services from the `_apps` collection.
```
Comment (Simran-B) [inline 3.5/programs-arangobackup-limitations.md]:
```suggestion
the state of the services from the `_apps` collection after a backup is restore.
```

File: 3.5/programs-arangobackup-limitations.md [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+instance which created the backup.
+
+Such an encrypted backup can only be restored to an instance using the
+same encryption key.
+
+Replication and Hot Backup
+--------------------------
+
+Hot backups are not automatically replicated between instances. This is
+true both for the active/failover setup with 2 (or more) single servers,
```
Comment (Simran-B) [inline 3.5/programs-arangobackup-limitations.md]:
```suggestion
true for both the active failover setup with 2 (or more) single servers
```

File: 3.5/programs-arangobackup-limitations.md [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+and for the data center to data center replication between clusters.
```
Comment (Simran-B) [inline 3.5/programs-arangobackup-limitations.md]:
```suggestion
and for the datacenter to datacenter replication between clusters.
```

File: 3.5/programs-arangobackup-limitations.md [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+Simply take hot backups on all instances.
```
Comment (Simran-B) [inline 3.5/programs-arangobackup-limitations.md]: Is there a chance of running into trouble if the passive side is behind the active side when both are restored?

File: 3.5/programs-arangobackup-limitations.md [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+indexes can take some time after the restore. It is planned to rectify
+this limitation in one of the next releases.
+
+Note furthermore that a running query with views can prevent a restore
+operation from happening whilst the query is running.
+
+Windows not Supported Yet
+-------------------------
+
+The first release of the hot backup feature (3.5.1) does not support
```
Comment (Simran-B) [inline 3.5/programs-arangobackup-limitations.md]:
```suggestion
The hot backup feature is not supported in the Windows version of ArangoDB
```

File: 3.5/programs-arangobackup-limitations.md (hunk 2 above)
Comment (Simran-B) [inline 3.5/programs-arangobackup-limitations.md]:
```suggestion
at this point in time.
```

Comment (Simran-B) [reply 3.5/programs-arangobackup-limitations.md]: Changed to not mention specific versions

**Comment by sleto-it** [comment]
@KVS85 because also the other tools have the limitation page inside the chapter that describe the tool. We can still link the same page from other sections

Hope this helps,

**Comment by Simran-B** [comment]
@sleto-it Only arangodump has a limitations chapter actually, so lets see how it would look like under the backup & restore chapter.

**Pull request merged**